ts/components/Lightbox.tsx

```diff
@@ -0,0 +1,141 @@
+/**
+ * @prettier
+ */
+import React from 'react';
+
+import classNames from 'classnames';
+
+interface Props {
+  close: () => void;
+  imageURL?: string;
+  onNext?: () => void;
+  onPrevious?: () => void;
+  onSave: () => void;
+  shouldShowNextButton: boolean;
+  shouldShowPreviousButton: boolean;
+  shouldShowSaveAsButton: boolean;
+}
+
+const styles = {
+  container: {
+    display: 'flex',
+    flexDirection: 'row',
+    position: 'absolute',
+    left: 0,
+    right: 0,
+    top: 0,
+    bottom: 0,
+    backgroundColor: 'rgba(0, 0, 0, 0.9)',
+    padding: 40,
+  } as React.CSSProperties,
+  objectContainer: {
+    flexGrow: 1,
+    display: 'inline-flex',
+    justifyContent: 'center',
+  } as React.CSSProperties,
+  image: {
+    flexGrow: 1,
+    flexShrink: 0,
+    maxWidth: '100%',
+    maxHeight: '100%',
+    objectFit: 'contain',
+  } as React.CSSProperties,
+  controls: {
+    flexShrink: 0,
+    display: 'flex',
+    flexDirection: 'column',
+    marginLeft: 10,
+  } as React.CSSProperties,
+};
+
+interface IconButtonProps {
+  type: 'save' | 'close' | 'previous' | 'next';
+  onClick?: () => void;
+}
+const IconButton = ({ onClick, type }: IconButtonProps) => (
+  <a href="#" onClick={onClick} className={classNames('iconButton', type)} />
+);
+
+export class Lightbox extends React.Component<Props, {}> {
+  public static defaultProps: Partial<Props> = {
+    shouldShowNextButton: false,
+    shouldShowPreviousButton: false,
+    shouldShowSaveAsButton: false,
+  };
+
+  private containerRef: HTMLDivElement | null = null;
+
+  public componentDidMount() {
+    const useCapture = true;
+    document.addEventListener('keyup', this.onKeyUp, useCapture);
+  }
+
+  public componentWillMount() {
+    const useCapture = true;
+    document.removeEventListener('keyup', this.onKeyUp, useCapture);
+  }
+
+  public render() {
+    const {
+      imageURL,
+      shouldShowNextButton,
+      shouldShowPreviousButton,
+      shouldShowSaveAsButton,
+    } = this.props;
+    return (
+      <div
+        style={styles.container}
+        onClick={this.onContainerClick}
+        ref={this.setContainerRef}
+      >
+        <div style={styles.objectContainer}>
+          <img
+            style={styles.image}
+            src={imageURL}
+            onClick={this.onImageClick}
+          />
+        </div>
+        <div style={styles.controls}>
+          <IconButton type="close" onClick={this.onClose} />
+          {shouldShowSaveAsButton ? (
```

Comment:
What if we only showed the button if we had a handler for it? That's what I did for `Quote` - it only shows the X button if a handler is provided for it.

Reply:
That’s one option, but the benefit of doing visibility separately from presence of a handler is that it allows more control in style guide without having to create dummy handlers.

Reply:
I'm just thinking about complexity of the problem domain/API here. Expressibility too - do we ever want to show the button without a handler? I don't think so.

Anyway, I have dummy handlers in the styleguide for `onClose` for `Quote` and it worked fine. Even allows me to log something to the console when the button is clicked. :0)